### PR TITLE
174945853 — rm AASM gem, update Publishable module

### PR DESCRIPTION
**Pull request by knowuh**
[#174945853]
https://www.pivotaltracker.com/story/show/174945853

### CODE REVIEW

File: commit_notes.txt [hunk 1]
```diff
@@ -0,0 +1,27 @@
+
```
Comment (dougmartin) [inline commit_notes.txt:1]: @knowuh did you mean to leave this file in the PR?

File: rails/spec/libs/publishable_spec.rb [hunk 2]
```diff
@@ -2,26 +2,85 @@
 
 require 'spec_helper'
 
+class DummyClass
```
Comment (dougmartin) [inline rails/spec/libs/publishable_spec.rb:5]: Do you still need this `aasm_column` column in the test class?

**Review by dougmartin — APPROVED** [review]
Looks good - just had a couple of questions.

File: rails/lib/publishable.rb [hunk 3]
```diff
@@ -1,37 +1,24 @@
...
       @@protected_publication_states=[:published]
       @@publication_states = [:draft,:published,:private]
-      @@publication_states.each { |s| aasm_state s}
 
       # this needs to come after the class variable definition...
       cattr_accessor :publication_states
 
-      aasm_event :publish do
-        transitions :to => :published, :from => [:draft]
+      def publish!
```
Comment (dougmartin) [inline rails/lib/publishable.rb:16]: I just looked at the test results and it looks like you also need to add a `publish` method:

```
undefined method `publish' for #<ExternalActivity:0x0000000006223a48>
Did you mean?  publish!
               published?
               public?
/home/travis/.rvm/gems/ruby-2.3.7/gems/activemodel-4.0.13/lib/active_model/attribute_methods.rb:439:in `method_missing'
/home/travis/.rvm/gems/ruby-2.3.7/gems/activerecord-4.0.13/lib/active_record/attribute_methods.rb:168:in `method_missing'
/home/travis/build/[secure]-consortium/rigse/rails/lib/mock_data/create_default_data.rb:661:in `block in create_default_study_materials'
```

**Review by dougmartin — CHANGES_REQUESTED** [review]
Need to implement `publish` method.

**Comment by knowuh** [comment]
@dougmartin thanks for the review,  I will clean those items up.

**Comment by knowuh** [comment]
@dougmartin  I think we are ready with this one.

**Review by dougmartin — APPROVED** [review]

**Pull request merged**